aero.minova.core.application.system.service/src/main/java/aero/minova/core/application/system/controller/XSqlProcedureController.java

```diff
@@ -0,0 +1,179 @@
+package aero.minova.core.application.system.controller;
+
+import java.sql.Connection;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import aero.minova.core.application.system.CustomLogger;
+import aero.minova.core.application.system.domain.ProcedureException;
+import aero.minova.core.application.system.domain.Row;
+import aero.minova.core.application.system.domain.SqlProcedureResult;
+import aero.minova.core.application.system.domain.Table;
+import aero.minova.core.application.system.domain.XSqlProcedureResult;
+import aero.minova.core.application.system.domain.XTable;
+import aero.minova.core.application.system.service.SecurityService;
+import aero.minova.core.application.system.sql.SystemDatabase;
+
+@RestController
+public class XSqlProcedureController {
+
+	@Autowired
+	SystemDatabase systemDatabase;
+
+	@Autowired
+	CustomLogger customLogger;
+
+	@Autowired
+	SecurityService securityService;
+
+	@Autowired
+	SqlProcedureController sqlProcedureController;
+
+	/**
+	 * Führt eine Liste von voneinander abhängenden SQL-Prozeduren aus. Die Prozeduren müssen in der richtigen Reihenfolge gesendet werden. Erst wenn alle
+	 * Prozeduren erfolgreich ausgeführt wurden, wird ein Commit getätigt. Tritt ein Fehler auf, werden alle bereits ausgeführten Prozeduren ge-rollbacked.
+	 *
+	 * @param inputTables
+	 *            Liste der Tables mit Namen der Prozeduren und deren Aufruf Parameter
+	 * @return Ergebnis Liste der Ergebnisse der Prozeduren-Aufrufe
+	 * @throws Exception
+	 *             Fehler bei der Ausführung von einer der Prozeduren. Rollback.
+	 */
+	@SuppressWarnings("unchecked")
+	@PostMapping(value = "data/x-procedure")
+	public ResponseEntity executeProcedures(@RequestBody List<XTable> inputTables) throws Exception {
+
+		customLogger.logUserRequest("data/x-procedure: ", inputTables);
+		List<XSqlProcedureResult> resultSets = new ArrayList<>();
+
+		final Connection connection = systemDatabase.getConnection();
```

Comment:
Die Connection, bitte in den try-Block verschieben.
Resourcen, die geschlossen oder sonst wie nach der Verarbeitung markiert werden müssen,
sollten immer in einen entsprechenden try-Block sich befinden.

Der Hintergrund ist der, dass 1ten immer etwas in der Zwischenzeit passieren kann
(OutOfMemoryError)
und 2ten, mit der Zeit wahrscheinlich zusätzlicher Code zwischen der Abholung des Verbindungs-Objektes
und den dazugehörigen try-block eingeschoben werden wird.